### PR TITLE
refactor: rename elementType to activeTool make it an object

### PR DESCRIPTION
**Pull request by ad1992**

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -1919,28 +1916,28 @@ class App extends React.Component<AppProps, AppState> {
     }
   });
 
-  private selectShapeTool(elementType: AppState["elementType"]) {
+  private selectShapeTool(activeTool: AppState["activeTool"]) {
```
Comment (dwelle) [inline src/components/App.tsx:1919]: let's rename this as well to `setActiveTool()`?

File: src/data/restore.ts [hunk 2]
```diff
@@ -30,8 +30,8 @@ type RestoredAppState = Omit<
   "offsetTop" | "offsetLeft" | "width" | "height"
 >;
 
-export const AllowedExcalidrawElementTypes: Record<
-  AppState["elementType"],
+export const AllowedExcalidrawactiveTools: Record<
```
Comment (dwelle) [inline src/data/restore.ts:33]:
```suggestion
export const AllowedExcalidrawActiveTools: Record<
```

File: src/element/resizeElements.ts [hunk 3]
```diff
@@ -186,13 +186,13 @@ const validateTwoPointElementNormalized = (
 };
 
 const getPerfectElementSizeWithRotation = (
-  elementType: string,
+  activeTool: string,
```
Comment (dwelle) [inline src/element/resizeElements.ts:189]: this is actually not a param for activeTool as we're passing excaldiraw elements, so `elementType` naming was correct

File: src/element/sizeHelpers.test.ts [hunk 4]
```diff
@@ -2,22 +2,22 @@ import { getPerfectElementSize } from "./sizeHelpers";
...
     expect(height).toEqual(140);
   });
-  it("should return height:0 if `elementType` is arrow and locked angle is 0", () => {
+  it("should return height:0 if `activeTool` is arrow and locked angle is 0", () => {
     const { height, width } = getPerfectElementSize("arrow", 200, 20);
     expect(width).toEqual(200);
     expect(height).toEqual(0);
   });
-  it("should return width:0 if `elementType` is arrow and locked angle is 90 deg (Math.PI/2)", () => {
+  it("should return width:0 if `activeTool` is arrow and locked angle is 90 deg (Math.PI/2)", () => {
```
Comment (dwelle) [inline src/element/sizeHelpers.test.ts:20]: ditto, `elementType` should stay

File: src/element/sizeHelpers.ts [hunk 5]
```diff
@@ -16,17 +16,17 @@ export const isInvisiblySmallElement = (
  * Makes a perfect shape or diagonal/horizontal/vertical line
  */
 export const getPerfectElementSize = (
-  elementType: string,
+  activeTool: string,
```
Comment (dwelle) [inline src/element/sizeHelpers.ts:19]: elementType

File: src/element/typeChecks.ts [hunk 6]
```diff
@@ -49,9 +49,9 @@ export const isFreeDrawElement = (
 };
 
 export const isFreeDrawElementType = (
-  elementType: ExcalidrawElement["type"],
+  activeTool: ExcalidrawElement["type"],
```
Comment (dwelle) [inline src/element/typeChecks.ts:52]: elementType

File: src/element/typeChecks.ts [hunk 7]
```diff
@@ -61,10 +61,10 @@ export const isLinearElement = (
 };
 
 export const isLinearElementType = (
-  elementType: AppState["elementType"],
+  activeTool: AppState["activeTool"],
```
Comment (dwelle) [inline src/element/typeChecks.ts:64]: elementType

File: src/element/typeChecks.ts [hunk 8]
```diff
@@ -75,9 +75,9 @@ export const isBindingElement = (
 };
 
 export const isBindingElementType = (
-  elementType: AppState["elementType"],
+  activeTool: AppState["activeTool"],
```
Comment (dwelle) [inline src/element/typeChecks.ts:78]: elementType

File: src/packages/excalidraw/README.md [hunk 9]
```diff
@@ -984,7 +984,7 @@ import { isLinearElement } from "@excalidraw/excalidraw";
 **Signature**
 
 <pre>
-isLinearElement(elementType?: <a href="https://github.com/excalidraw/excalidraw/blob/master/src/element/types.ts#L80">ExcalidrawElement</a>): boolean
+isLinearElement(activeTool?: <a href="https://github.com/excalidraw/excalidraw/blob/master/src/element/types.ts#L80">ExcalidrawElement</a>): boolean
```
Comment (dwelle) [inline src/packages/excalidraw/README.md:987]: elementType

File: src/packages/excalidraw/README_NEXT.md [hunk 10]
```diff
@@ -990,7 +990,7 @@ import { isLinearElement } from "@excalidraw/excalidraw-next";
 **Signature**
 
 <pre>
-isLinearElement(elementType?: <a href="https://github.com/excalidraw/excalidraw/blob/master/src/element/types.ts#L80">ExcalidrawElement</a>): boolean
+isLinearElement(activeTool?: <a href="https://github.com/excalidraw/excalidraw/blob/master/src/element/types.ts#L80">ExcalidrawElement</a>): boolean
```
Comment (dwelle) [inline src/packages/excalidraw/README_NEXT.md:993]: elementType

File: src/element/dragElements.ts [hunk 11]
```diff
@@ -92,7 +92,7 @@ export const getDragOffsetXY = (
 
 export const dragNewElement = (
   draggingElement: NonDeletedExcalidrawElement,
-  elementType: AppState["elementType"],
+  activeTool: AppState["activeTool"],
```
Comment (dwelle) [inline src/element/dragElements.ts:95]: we've had it wrong here, it should be

```
elementType: ExcalidrawElement["type"]
```

File: src/components/App.tsx [hunk 12]
```diff
@@ -3573,14 +3570,14 @@ class App extends React.Component<AppProps, AppState> {
...
-        elementType: "selection",
+        activeTool: "selection",
       });
     }
   };
 
   private handleFreeDrawElementOnPointerDown = (
     event: React.PointerEvent<HTMLCanvasElement>,
-    elementType: ExcalidrawFreeDrawElement["type"],
+    activeTool: ExcalidrawFreeDrawElement["type"],
```
Comment (dwelle) [inline src/components/App.tsx:3580]: elementType

File: src/element/sizeHelpers.ts (hunk 5 above)
Comment (ad1992) [reply src/element/sizeHelpers.ts:19]: downside of `cmd+shift+h` 🤦‍♀️

**Review by dwelle — APPROVED** [review]

**Pull request merged**